### ISSUE TITLE
PyQt5

### ISSUE THREAD

**Issue by pedromorgan** [ShadowKyogre/mikidown#56]
- Are there any plans to port to Qt5 (can I help) ??
- Also making it work on windows ?

**Comment by ShadowKyogre** [comment]
For your first bullet, the answer'd be yes. I would appreciate the extra help since I'm juggling coding this with several other things.

For your second bullet, it does work on Windows the last time I tried it (but I don't have it well-documented yet). The way I did it was running the main python script directly in the folder. Could you elaborate on what you're looking for in making-it-work-for-Windows? (Like being able to run it w/o a full-blown installation of python or something like that.)

**Comment by pedromorgan** [comment]
Re Qt5, I recently ported [ts2](https://github.com/ts2/ts2)

One of the features was to include a single `Qt.py` file  for "importing qt", with the idea that later pyside3 etc could be used. The idea was ripped from pyqt graph
- https://github.com/ts2/ts2/blob/master/Qt.py#L21
- https://github.com/pyqtgraph/pyqtgraph/blob/develop/pyqtgraph/Qt.py

This means code would look like
```py
from Qt import QtCore, QtWidgets, Qt

foo = QtWidgets.QMainWindow()
```

Also I personally dont like importing every "class" at the top,..(but thats me ;-))
```py
from PyQt4.QtGui import QSyntaxHighlighter, QColor, QFont, QTextCharFormat

class Foo(QtWidget):

#vs
from Qt import QtGui

class Foo(QtGui.QtWidget):
```

Your comments would be appreciated before i start :-)

**Comment by pedromorgan** [comment]
Disclosure: Hidden agenda is to expreiment with embedding the "mikidown editor" within other applications eg TS2

**Comment by pedromorgan** [comment]
WIP (DONT MERGE)
https://github.com/ShadowKyogre/mikidown/pull/58

**Comment by pedromorgan** [comment]
Have had no feedback, or comments... is there anyone out there ;-)))

**Comment by ShadowKyogre** [comment]
Sorry about the long delay ><. Was floundering around for quite a bit (and still am) trying to build up skills to help get a job (on top of also juggling creative writing to help pull in money in the mean time). Not exactly sure where to put the notice about my status since my online presence is kind of scattered.

I ran the port today and so far notice no problems. However, I'm also reading through the rest of the code changes right now to make sure I didn't miss anything.

**Comment by pedromorgan** [comment]
cool stuff..
Kinda need this for a new project.. so if u go with it, I can probably maintain the "bits" i need.. ;-)
eg git integration..

If your ok with things, then I keep hacking

**Comment by pedromorgan** [comment]
ohhh.. and target is windows7+.. and am developing with a penguin...

**Comment by ShadowKyogre** [comment]
@pedromorgan : Git integration as in being able to view the history on a per-note basis (as well as per-commit), where the notebook is the git repo (just as an example of such integration)? That'd definitely be interesting to see implemented. I've toyed around with viewing the history on a per-file basis for my own story stuff, but haven't gotten any code for it for Mikidown. I'd love to see this.

And that's all right xD. My main development and writing machine's a penguin too, though I do use Windows 7 too.

**Comment by pedromorgan** [comment]
In my case...
instead of notebooks (and a previous zim user and dev.. from yonder year)..
their chapters in a book, or sections in a website.. and both..

So a workspace is in.. and similar...
Am also working on the server side, and push to deploy, eg on OpenShift/AppEngine and alike

**Comment by ShadowKyogre** [comment]
@pedromorgan : Understood! That'd also be a good thing to develop it for. Do lemme know when you have some code for that.

**Comment by pedromorgan** [comment]
In fact the "terminology of labels".. is strange, particular when your on the front end with writers etc..

Have u looked at gitbooks etc.. they are real cool.
But unlike mikidown, "its not a desktop" and main thing is
CONTEXT switch.. for end users...

That is why am pursuing this project, and kinda want it as a "markdown-plugin" editor/viewer
embedded in an application...
No more work documents, n more PDF, even the Md's secretarys son can fix
a spelling mishtake.. even across the network.. 

So this is cool, am happy.. and ..
opening 2 beers. one for @ShadowKyogre  and me..
oops he's not here'' I better consume instead lol ;-)

**Comment by ShadowKyogre** [comment]
lol xD at the last paragraph. 

And yeah, I've looked at gitbooks in the past. Looking at it again, it definitely looks like it's leapt bounds from the last time I saw it (when it only used markdown).The only reason I haven't used it yet is due to the nature of the stuff that I write for story stuff. (<- those require some privacy even after pushing changes remotely to make sure it's backed up). 

But I definitely do agree it's a pretty big context switch. And their desktop editor isn't open source too.
